platform/macos/detect.py

```diff
@@ -242,10 +242,17 @@ def configure(env: "SConsEnvironment"):
         env.Append(LINKFLAGS=["-framework", "Metal", "-framework", "IOSurface"])
         if not env["use_volk"]:
             env.Append(LINKFLAGS=["-lMoltenVK"])
-            mvk_path = detect_mvk(env, "macos-arm64_x86_64")
+
+            mvk_path = ""
+            for arch in ["macos-arm64_x86_64", "macos-arm64"]:
+                mvk_path = detect_mvk(env, arch)
+                if mvk_path == "":
+                    continue
+
+                mvk_path = os.path.join(mvk_path, arch)
```

Comment:
The loop should break as soon as it's found, or it will go over all options and override path with empty string on next iteration.

Also, on Intel Mac, homebrew will probably have `macos-x86_64` subfolder. So it should look for correct arch (and universal `macos-arm64_x86_64`) only, based on `env["arch"]` value (which can be `x86_64` or `arm64`).

Reply:
> The loop should break as soon as it's found, or it will go over all options and override path with empty string on next iteration.

Gah, you are absolutely right. Thanks for pointing that out. I was trying to avoid nesting. 🤦 

> Also, on Intel Mac, homebrew will probably have macos-x86_64 subfolder. So it should look for correct arch (and universal macos-arm64_x86_64) only, based on env["arch"] value (which can be x86_64 or arm64).

Are you suggesting to populate the _arch variants_ array based on the `env["arch"]` value?  
I would find it simpler to just add `macos-x86_64` to the _arch variants_ array.

Reply:
> I would find it simpler to just add macos-x86_64 to the arch variants array.

It will work with homebrew, but it might select the wrong arch if you have a custom build with both `macos-x86_64` and `macos-arm64` (this unusably should not be the case).

Reply:
Okay, I am not sure, whether we should include the universal _variant_ when `env["arch"] == "x86_64"`?

Reply:
> Okay, I am not sure, whether we should include the universal variant when env["arch"] == "x86_64"?

Why not, universal variant should work with both arm64 and x86_64 (and is the only one used ns the official MoltenVK builds).

Reply:
Alright, I have tried to implement your suggestions. Thanks for the hints!
Please have a look at the changes.